packages/back-end/src/controllers/stripe.ts

```diff
@@ -130,6 +153,36 @@ export async function getSubscriptionQuote(req: AuthRequest, res: Response) {
   });
 }
 
+export async function getHasValidPaymentMethod(
+  req: AuthRequest,
+  res: Response
+) {
+  req.checkPermissions("manageBilling");
+
+  if (!IS_CLOUD) {
+    return res.status(200).json({
+      status: 200,
+      hasValidPaymentMethod: null,
+    });
+  }
+
+  const { org } = getOrgFromReq(req);
+
+  if (!org.stripeCustomerId) {
+    throw new Error("Missing customer id");
+  }
+
+  const paymentMethods = await stripe.paymentMethods.list({
+    customer: org.stripeCustomerId,
+    type: "card",
+  });
```

Comment:
Instead of a new Stripe API call, can we just add a new property to `organization.subscription` that says whether or not the subscription has a payment method attached?

Reply:
Added this for webhooks as a separate fetch from Stripe. The needed data isn't in the webhook body unfortunately.